client/matching/weighted_loadbalancer.go

```diff
@@ -165,8 +166,8 @@ func (lb *weightedLoadBalancer) PickReadPartition(
 	if !ok {
 		return lb.fallbackLoadBalancer.PickReadPartition(domainID, taskList, taskListType, forwardedFrom)
 	}
-	p := w.pick()
-	lb.logger.Debug("pick read partition", tag.WorkflowDomainID(domainID), tag.WorkflowTaskListName(taskList.GetName()), tag.WorkflowTaskListType(taskListType), tag.Dynamic("weights", w.weights), tag.Dynamic("tasklist-partition", p))
```

Comment:
w.weights cannot be accessed without lock being held, so update the pick method to return cumulativeWeights for debugging